prboom2/src/SDL/i_sound.c

```diff
@@ -227,6 +227,23 @@ static int addsfx(int sfxid, int channel, const unsigned char *data, size_t len)
   return channel;
 }
 
+static int getBufferSize(void)
+{
+  int limit, i;
+
+//if (snd_samplecount) return snd_samplecount;
+
+  limit = snd_samplerate / TICRATE;
+
+  // adapted from Chocolate Doom
+  // gets the smallest power of 2, for feeding into SDL
+  for (i = 0;; ++i)
+  {
+    if ((1 << (i + 1)) > limit)
+      return (1 << i);
+  }
+}
```

Comment:
Please keep the comments of the original code (and the exit strategy).

Reply:
Newest commit is (mostly) as is in Chocolate.